app/services/association_authorizer.rb

```diff
@@ -0,0 +1,27 @@
+class AssociationAuthorizer
+  def self.authorized_associations(associations,  klass_name = nil, should_raise_exception = true)
+    if associations.included_modules.include?(Authorizable)
+      associations_klass = associations
+      if associations.respond_to?(:klass)
+        associations_klass = associations.klass
+      end
+      klass_name ||= associations_klass
+      permission = view_permission_name(klass_name, should_raise_exception)
+      associations.authorized(permission, associations_klass) if permission
+    else
+      associations
+    end
+  end
+
+  def self.view_permission_name(klass, should_raise_exception)
+    permission = "view_#{klass.to_s.underscore.pluralize}"
+    unless Permission.where(:name => permission).present?
+      if should_raise_exception
+        raise Foreman::Exception.new(N_('unknown permission %s'), permission)
+      else
+        false
```

Comment:
I don't think this is going to do anything, since the return value is determined by the `permission` on line 25.  Can you add a test to ensure the return value is consistent with both values of should_raise_exception?